compiler/z/codegen/OMRMemoryReference.cpp

```diff
@@ -2940,282 +2935,121 @@ OMR::Z::MemoryReference::generateBinaryEncoding(uint8_t * cursor, TR::CodeGenera
    TR::Node * node = instr->getNode();
 
    bool is2ndSSMemRef=self()->is2ndMemRef();
-   bool useNodesForLongFixup = !comp->getOption(TR_DisableLongDispNodes);
-
-   /*
-      RX Format
-       ________ ____ ____ ____ ____________
-      |Op Code | R1 | X2 | B2 |     D2     |
-      |________|____|____|____|____________|
-      0         8   12   16   20          31
-     - or -
-      RS Format (LM and STM)
-       ________ ____ ____ ____ ____________
-      |Op Code | R1 | R3 | B2 |     D2     |
-      |________|____|____|____|____________|
-      0         8   12   16   20          31
-     - or -
-      RXE Format
-       ________ ____ ____ ____ __ ________ ____________ _______
-      |Op Code | R1 | X2 | B2 |D2         | ////////// |Op Code|
-      |________|____|____|____|___________|____________|_______|
-      0         8   12   16   20          32           40      47
-     - or -
-      RXF Format
-       ________ ____ ____ ____ __ ________ ___ _________ _______
-      |Op Code | R1 | X2 | B2 |D2         | R1 | ////// |Op Code|
-      |________|____|____|____|___________|___ _________|_______|
-      0         8   12   16   20          32   36      40      47
-     - or -
-      RXY Format
-       ________ ____ ____ ____ __ ________ ____________ _______
-      |Op Code | R1 | X2 | B2 |DL2        | DH2        |Op Code|
-      |________|____|____|____|___________|____________|_______|
-      0         8   12   16   20          32           40      47
-    */
-
-
-
-   //calculate an estimate for the displacment of this reference
-   disp = self()->calcDisplacement(cursor, instr, cg);
+
+   int32_t displacement = self()->calcDisplacement(cursor, instr, cg);
 
    //add project specific relocations for specific instructions
-   self()->addInstrSpecificRelocation(cg, instr, disp, cursor);
+   self()->addInstrSpecificRelocation(cg, instr, displacement, cursor);
+
+   // We may end up "upgrading" the instruction in case of long displacement, so this variable may change
+   auto instructionFormat = instr->getKind();
 
-   // Most cases we will know at inst selection phase that the disp is big.  But it is still possible
-   // for us to hit bin phase and realize the DISP is large (eg when the stack is large).
-   if (disp<0 || disp>=MAXDISP)
+   // In most cases we will know at instruction selection that the displacement is large. But it is still possible for
+   // us to reach binary encoding without knowing a displacement will be large. A typical example of this is when 
+   // referencing stack symbols whose displacement will be unknown until the stack is mapped; an operation that happens
+   // after instruction selection and before binary encoding.
+   if (displacement < 0 || displacement >= MAXDISP)
```

Comment:
Maybe we can use the newly added `MINDISP` instead of a 0?

Reply:
Good catch. Thanks! Fixed in 0514fc6